nodejs/eks/nodegroup.ts

```diff
@@ -1867,24 +1880,26 @@ export function isGravitonInstance(instanceType: string): boolean {
  * architecutre of the instance type.
  */
 function getAMIType(
-    amiType: pulumi.Input<string> | undefined,
-    gpu: pulumi.Input<boolean> | undefined,
-    instanceType: pulumi.Input<string> | undefined,
-): string {
-    if (amiType) {
-        // Return the user-specified AMI type.
-        return amiType.toString();
-    }
+    amiType: pulumi.Input<string> | pulumi.Output<string> | undefined,
```

Comment:
`pulumi.Input<string> | pulumi.Output<string> | undefined` I *think* defines the same type space as ` `pulumi.Input<string> | undefined` because `pulumi.Input<string>` expands to `pulumi.Output<string> | Promise<string> | string`. Perhaps this is not needed?

Reply:
Yes, that is correct! Removed.